src/java/picard/sam/markduplicates/util/OpticalDuplicateFinder.java

```diff
@@ -72,9 +72,12 @@ public OpticalDuplicateFinder(final String readNameRegex, final int opticalDupli
      * unmodified a copy of the list should be passed to this method.
      *
      * @param list a list of reads that are determined to be duplicates of one another
+     * @param keeper a single PhysicalLocation that is the one being kept as non-duplicate, and thus should never be
+     *               annotated as an optical duplicate. May in some cases be null, or a PhysicalLocation not
+     *               contained within the list!
```

Comment:
It's because, to do the optical duplicate detection, you need to see all the locations, including the read pair that is being kept as non-duplicate.  I.e. if you have two read pairs with same alignment positions, you need both to determine if one is an optical duplicate of the other, but you should only mark one of the with an optical duplicate tag.  And ideally *not* the one that isn't being marked as a duplicate.